app/controllers/works_controller.rb

```diff
@@ -34,7 +39,12 @@ def create
   end
 
   def show
-    @votes = @work.votes.order(created_at: :desc)
+    if @login_user
+      @votes = @work.votes.order(created_at: :desc)
+    else
+      flash[:result_text] = "You must log in to do that"
```

Comment:
This is a good start on authorization, but you would have to repeat this work for every controller action. A better approach might be to use a `before_action` in `ApplicationController`, then turn it off for specific actions with `skip_before_action`.